### PR TITLE
[JENKINS-61930] JCasC testing

### PR DESCRIPTION
**Pull request by Vlatombe**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -53,6 +53,7 @@
     <pipeline-model-definition.version>1.6.0</pipeline-model-definition.version>
     <slf4jVersion>1.7.26</slf4jVersion>
     <useBeta>true</useBeta>
+    <configuration-as-code.version>1.36</configuration-as-code.version>
```
Comment (jglick) [inline pom.xml:56]: Better to use the BOM.

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/casc/CasCTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        podTemplate = templates.get(1);
+        assertEquals(false, podTemplate.isHostNetwork());
+        assertEquals("dynamic-pvc", podTemplate.getLabel());
+        assertEquals("dynamic-pvc", podTemplate.getName());
+        assertThat(podTemplate.getYamlMergeStrategy(), isA(Overrides.class));
+        WorkspaceVolume workspaceVolume = podTemplate.getWorkspaceVolume();
+        assertNotNull(workspaceVolume);
+        assertThat(workspaceVolume, isA(DynamicPVCWorkspaceVolume.class));
+        DynamicPVCWorkspaceVolume dynamicPVCVolume = (DynamicPVCWorkspaceVolume) workspaceVolume;
+        assertEquals("ReadWriteOnce", dynamicPVCVolume.getAccessModes());
```
Comment (jglick) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/casc/CasCTest.java:48]: OK I am bored already. :sleepy:

File: src/test/resources/org/csanchez/jenkins/plugins/kubernetes/casc/casc_envVar_keyValue.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+jenkins:
+  clouds:
+    - kubernetes:
+        containerCapStr: "10"
```
Comment (jglick) [inline src/test/resources/org/csanchez/jenkins/plugins/kubernetes/casc/casc_envVar_keyValue.yaml:4]: Ugly. We cannot convert to type `Integer`?

File: pom.xml (hunk 1 above)
Comment (kerogers-cloudbees) [inline pom.xml]: They're up to 1.39 now.

Comment (Vlatombe) [reply pom.xml]: I'll just pick up whatever is in BOM, I don't think we need anything particular that is in 1.39

**Comment by Vlatombe** [comment]
> Can you explain where the yaml is being loaded into the config for the tests?

This is handled by `RoundTripAbstractTest` which is picking up the return value of `configResource()`

**Pull request merged**